### PR TITLE
Cellular docs updated

### PR DESCRIPTION
**Pull request by AriParkkila**
Cellular docs updated for Mbed OS 5.12.

Porting guide need to be finished later.

### CODE REVIEW

File: docs/api/networkinterfaces/CellularInterface.md [hunk 1]
```diff
@@ -69,6 +67,6 @@ This example establishes connection with the cellular network using Mbed OS Cell
 ### Related content
 
 - [Network socket](network-socket.html) API reference overview.
-- [Cellular architecture](../reference/cellular.html).
+- [Cellular overview](../reference/cellular.html) describes more details about the cellular architecture and its usage.
```
Comment (melwee01) [inline docs/api/networkinterfaces/CellularInterface.md]: describes cellular architecture and its usage in more detail.

File: docs/porting/connectivity/CellularInterface.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
-## Cellular module porting
+## Cellular device porting
 
-This document provides guidelines how to make a cellular modem adaptation for Mbed OS. Please see [Cellular API usage](../apis/cellular-api.html) about how to use cellular modules from an application point of view.
+This document provides guidelines how to make a cellular device adaptation for Mbed OS.
 
-### Adding modem target support
+### Adding cellular on Mbed OS target
```
Comment (melwee01) [inline docs/porting/connectivity/CellularInterface.md]: on an Mbed OS target

File: docs/porting/connectivity/CellularInterface.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ typedef enum {
 
 ```
 
-### Adding a new cellular target
+### Adding a new cellular device
 
 You need to specify in [CellularTargets.h](https://os.mbed.com/docs/development/mbed-os-api-doxy/_cellular_targets_8h_source.html) the `<manufacturer-module>` that is mounted on your board.
```
Comment (melwee01) [inline docs/porting/connectivity/CellularInterface.md]: ~that is~ mounted

**Review by melwee01 — CHANGES_REQUESTED** [review]
Bit too much passive voice for my taste, but I don't know that we can afford to be that picky.

File: docs/api/networkinterfaces/CellularInterface.md [hunk 4]
```diff
@@ -1,10 +1,8 @@
-<h2 id="cellular-api">Cellular</h2>
+<h2 id="cellular-api">Cellular API</h2>
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md:1]: Query: Why did we add "API" here?

File: docs/api/networkinterfaces/CellularInterface.md [hunk 5]
```diff
@@ -1,10 +1,8 @@
...
 
-<span class="images">![](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_base.png)<span>CellularBase class hierarchy</span></span>
+<span class="images">![](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_interface.png)<span>CellularInterface class hierarchy</span></span>
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md:3]: Query: Where should this link go? I don't see any class in the Doxygen called "CellularInterface".

File: docs/api/networkinterfaces/CellularInterface.md [hunk 6]
```diff
@@ -1,10 +1,8 @@
...
 
-The [CellularBase](https://os.mbed.com/docs/development/mbed-os-api-doxy/_cellular_base_8h_source.html) provides a C++ API for connecting to the internet over a Cellular device.
-
-Arm Mbed OS provides a [reference implementation of CellularBase](https://os.mbed.com/docs/development/mbed-os-api-doxy/_cellular_context_8h_source.html), which has more information.
+The [CellularInterface](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_interface.html) class provides a C++ API for connecting to the internet over a Cellular device.
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md:5]: Query: Where should this link go? It 404s for me.

File: docs/api/networkinterfaces/CellularInterface.md [hunk 7]
```diff
@@ -48,15 +46,15 @@ You can use and extend a cellular interface in various different ways. For examp
 - The easy cellular connection uses standard 3GPP AT 27.007 AT commands to set up the cellular modem and to register to the network.
 - After registration, the driver opens a PPP pipe using LWIP with the cellular modem and connects to the internet. If AT only mode is in use, then modem-specific AT commands are used for socket data control.
 
-### CellularBase class reference
+### CellularInterface class reference
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md:49]: Do we have a class called CellularInterface? I can't find it in the Doxygen.

File: docs/api/networkinterfaces/CellularInterface.md [hunk 8]
```diff
@@ -48,15 +46,15 @@ You can use and extend a cellular interface in various different ways. For examp
...
 
-[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_base.html)
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_interface.html)
```
Comment (AnotherButler) [inline docs/api/networkinterfaces/CellularInterface.md:51]: Where should this link go? It 404s for me.

File: docs/porting/connectivity/connectivity.md [hunk 9]
```diff
@@ -12,7 +12,7 @@ The **DeviceInterface** should also inherit one of the following (unless it is a
 
 - [EthInterface](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_eth_interface.html).
 - [WiFiInterface](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_wi_fi_interface.html).
-- [CellularInterface](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_base.html).
+- [CellularInterface](https://os.mbed.com/docs/development/mbed-os-api-doxy/class_cellular_interface.html).
```
Comment (AnotherButler) [inline docs/porting/connectivity/connectivity.md:15]: Query: Where should this link go? It 404s for me.

File: docs/reference/technology/connectivity/cellular.md [hunk 10]
```diff
@@ -12,49 +12,30 @@ Key features of the Mbed OS cellular APIs include:
 
 - Compatible with 3GPP TS 27.007 and 27.005.
 - Implements core functionality for OMA Lightweight M2M Client.
-- Hosted cellular modules on Mbed Enabled boards.
+- Quick start with Mbed Enabled boards.
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: Is the quick start a key features of the Mbed OS cellular APIs? That doesn't sound right to me.

File: docs/reference/technology/connectivity/cellular.md [hunk 11]
```diff
@@ -12,49 +12,30 @@ Key features of the Mbed OS cellular APIs include:
...
 
 OMA Lightweight M2M is a protocol from the Open Mobile Alliance for IoT device management. The Mbed OS cellular API provides core functionality to implement the LWM2M Client. For more information, please see _OMA LightweightM2M_ and _OMA LWM2M Object Connectivity Management_ at [OMA Specifications](http://openmobilealliance.org/wp/index.html).
 
-Mbed OS already supports several Mbed Enabled boards with on-board cellular hosted modules. Because Mbed OS is an open source platform, developers can enable support for new cellular boards with our adaptation framework. Please see our [cellular porting guide](../porting/porting-connectivity.html#cellularinterface) for more information.
+Many Mbed Enabled boards already support cellular connectivity. Because Mbed OS is an open source platform, developers can enable support for new cellular boards with our adaptation framework. Please see our [cellular porting guide](https://os.mbed.com/docs/v5.11/porting/cellular-module-porting.html) for more information.
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: Please change link to "(../porting/cellular-module-porting.html)", so it's not tied to a specific version.

File: docs/reference/technology/connectivity/cellular.md [hunk 12]
```diff
@@ -12,49 +12,30 @@ Key features of the Mbed OS cellular APIs include:
...
+With cellular, the easiest way to connect your application to the internet over a cellular network is to use the `CellularContext` class and `get_default_instance`. It encapsulates most of the complexity of connecting to the cellular network and also reports any changes in connection status to your application. When connected to a cellular network, you can use Mbed OS network sockets as usual.
 
-<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/api-cellular-quick-start.png)<span>Figure 2. Connect to cellular network and open a socket</span></span>
+A network interface instantiated directly or using calls such as `CellularInterface::get_default_instance()` or `CellularContext::get_default_instance()` is initially unconfigured. You can call `NetworkInterface::set_default_parameters()` to set the default parameters that would have been set if the interface had been requested using `NetworkInterface::get_default_instance()` (see nsapi JSON configuration).
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: When we say "see nsapi JSON configuration", is there a link we can add?

File: docs/reference/technology/connectivity/cellular.md [hunk 13]
```diff
@@ -12,49 +12,30 @@ Key features of the Mbed OS cellular APIs include:
...
-            "CELLULAR_DEVICE=<manufacturer-module>",
-            "MDMRXD=<rx-pin-name>",
-            "MDMTXD=<tx-pin-name>",
-            "MDMRTS=<rts-pin-name>",
-            "MDMCTS=<cts-pin-name>"
-        ]
-    }
-
-You need to change the pin names above to actual pins, such as D0 and D1, according to your Mbed target. You may also need to define MDMRTS and MDMCTS pins if you have RTS and CTS connected on UART. If RTC and CTS are not connected on UART, then define MDMRTS and MDMCTS as `NC`.
+If Mbed OS target board has an on-board cellular device then the cellular framework uses that by default when calling `get_default_instance()`. The default cellular device instance can be overridden in the cellular driver `mbed_lib.json` files or an application can override  `NetworkInterface::get_default_instance()`.
```
Comment (AnotherButler) [inline docs/reference/technology/connectivity/cellular.md]: "If an Mbed ..."

**Review by AnotherButler — COMMENTED** [review]
Thanks for the PR. I left some comments asking about links I can't get to work. I think maybe that's on the Doxygen/code end?

File: docs/api/networkinterfaces/CellularInterface.md (hunk 4 above)
Comment (AriParkkila) [reply docs/api/networkinterfaces/CellularInterface.md:1]: It's to indicate that this page is for API developers, I guess that's right..?

File: docs/api/networkinterfaces/CellularInterface.md (hunk 5 above)
Comment (AriParkkila) [reply docs/api/networkinterfaces/CellularInterface.md:3]: Mbed OS 5.12 will rename CellularBase => CellularInterface.

File: docs/api/networkinterfaces/CellularInterface.md (hunk 6 above)
Comment (AriParkkila) [reply docs/api/networkinterfaces/CellularInterface.md:5]: See above.

File: docs/api/networkinterfaces/CellularInterface.md (hunk 7 above)
Comment (AriParkkila) [reply docs/api/networkinterfaces/CellularInterface.md:49]: See above.

File: docs/api/networkinterfaces/CellularInterface.md (hunk 8 above)
Comment (AriParkkila) [reply docs/api/networkinterfaces/CellularInterface.md:51]: See above.

File: docs/porting/connectivity/connectivity.md (hunk 9 above)
Comment (AriParkkila) [reply docs/porting/connectivity/connectivity.md:15]: See above.

File: docs/reference/technology/connectivity/cellular.md (hunk 10 above)
Comment (AriParkkila) [reply docs/reference/technology/connectivity/cellular.md]: Could be written also that "Supports Mbed Enabled boards"..?

**Comment by AnotherButler** [comment]
@AriParkkila Thanks for clarifying. Because this depends on code that hasn't merged yet, I'm going to add a code dependency label. Could you please link to the relevant mbed-os PR?

**Comment by AnotherButler** [comment]
@AriParkkila  or @mirelachirica Could one of you please let me know which code PR (if there is one) this is waiting for? Also, could you please resolve the merge conflicts?

**Comment by AriParkkila** [comment]
@AnotherButler PR has not been made yet. This time we documented it before anything was implemented, so I guess we are learning :) I will rebase and update this when PR is ready.

**Comment by AriParkkila** [comment]
@AnotherButler PR for the CellularInterface class https://github.com/ARMmbed/mbed-os/pull/9746

Rebased and ready for merge.

**Pull request merged**